### PR TITLE
Lesson defs

### LINKED ISSUE #23: Lesson TypeDefs

### PR DESCRIPTION
**Pull request by sarah-jensen**
Testing should work with updated seeds.

Query: view single lesson by ID

Mutation: add lesson, update lesson (any field except ID can be edited), delete lesson

closes #23 #22

### CODE REVIEW

File: server/schemas/lesson.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  extend type Query {
+    lesson(id: ID!): Lesson
+    lessons: [Lesson]
+  }
+
+  extend type Mutation {
+    addLesson(name: String!, body: String!, media: String, duration: Int!): Lesson
+    updateLesson(lessonId: ID!, name: String, body: String media: String, duration: String): Lesson
```
Comment (sendusyourbones) [inline server/schemas/lesson.js]: I believe this should be `duration: Int` rather than `duration: String`

Comment (sarah-jensen) [reply server/schemas/lesson.js]: you are absolutely correct, fixing now!

**Comment by sendusyourbones** [comment]
This PR has `server/schemas/lesson.js` and `server/schemas/lessonDefs.js` which I think are the same thing. I know we talked about renaming the files from the `defs` versions and I neglected to do that in my PRs. For this PR approval please pick one file and delete the other. Then let me know if I should update any other file names!

File: server/schemas/lesson.js (hunk 1 above)
Comment (sarah-jensen) [reply server/schemas/lesson.js]: submitted new commit with correction

File: server/schemas/lessonDefs.js [hunk 2]
```diff
@@ -2,28 +2,78 @@ const { gql } = require('apollo-server-express');
...
+    //get a single lesson by id
+    lesson: 
+        async (parent, { _id }) => {
+            return await Lesson.findById(_id);
+        },
+  },
+
+  Mutation: {
+    //ADD a new lesson
+    addLesson: async (parent, { name, body, media, duration }) => {
```
Comment (sendusyourbones) [inline server/schemas/lessonDefs.js]: I believe this should be modified to include adding the lesson to the tutorial after the lesson is added to the db. I did something similar in my `reviewDefs` if you want to build from there.

Comment (sarah-jensen) [reply server/schemas/lessonDefs.js]: that's a good idea, I'll work on that now

Comment (sarah-jensen) [reply server/schemas/lessonDefs.js]: pushed new commit

**Review by sendusyourbones — APPROVED** [review]
Works great!

**Pull request merged**